server/src/main/kotlin/com/subject/server/repository/dsl/SearchCondition.kt

```diff
@@ -0,0 +1,20 @@
+package com.subject.server.repository.dsl
+
+import com.querydsl.core.types.dsl.BooleanExpression
+import com.subject.server.domain.QPatient.patient
+
+enum class SearchCondition {
+    NAME, BIRTHDAY, REGISTRATION;
+
+    // 리팩토링이 필요해보입니다
+    fun getPatientBooleanExpressionByKeyword(keyword: String?): BooleanExpression? {
+        if (this == NAME)
+            return patient.name.eq(keyword)
+        if (this == BIRTHDAY)
+            return patient.birthday.eq(keyword)
+        if (this == REGISTRATION)
+            return patient.registrationNumber.eq(keyword)
+
+        return null
+    }
```

Comment:
테스트는 안해봤는데 이렇게 리팩토링 해볼 수 있지 않을까 생각합니다.  
이넘 클래스안에 dsl 관련 의존성이 생기는게 조금 별로인거 같기도 한데요... 
실제 사용하는 곳에서 private 한 확장함수로 사용해도 괜찮을거 같습니다~
```suggestion
    enum class SearchCondition(
    val expression: (String?) -> BooleanExpression
) {
    NAME({ keyword -> patient.name.eq(keyword) }),
    BIRTHDAY({ keyword -> patient.birthday.eq(keyword) }),
    REGISTRATION({ keyword -> patient.registrationNumber.eq(keyword) });

    // 리팩토링이 필요해보입니다
    fun getPatientBooleanExpressionByKeyword(keyword: String?) = this.expression(keyword)
}
```

Reply:
[b967ea](https://github.com/jhjeong-hdj/subject/tree/b967ea54d74af812f6735273c0987c7a8bf7d0b3)